lib/Mojo/Util.pm

```diff
@@ -248,6 +249,22 @@ sub secure_compare {
   return $r == 0;
 }
 
+sub slugify {
+  my ($value, $allow_unicode) = @_;
+  if ($allow_unicode) {
+
+    # force unicode semantics by upgrading string
+    utf8::upgrade($value = Unicode::Normalize::NFKC($value));
+    $value =~ s/[^\w\s-]+//g;
+  }
+  else {
+    $value = Unicode::Normalize::NFKD($value);
+    $value =~ s/[^a-zA-Z0-9_\p{PosixSpace}-]+//g;  # ascii word/space/dash chars
```

Comment:
\p{PosixSpace} may be replaced with the equivalent ` \t\n\cK\f\r` but I'm not sure if that's clearer.